### ISSUE TITLE
Docker Hub is sunsetting free team accounts

### ISSUE THREAD

**Issue by dsander** [huginn/huginn#3234]
Hi,

I got an email from Docker Hub today telling me that they are [sunsetting free team account](https://web.docker.com/rs/790-SSB-375/images/privatereposfaq.pdf?utm_campaign=2023-03-14-operational-team-discontinued-ft&utm_medium=email&utm_source=marketo&mkt_tok=NzkwLVNTQi0zNzUAAAGKf9wMD34iwIUJJNAWOwJt52o9JzqOjjQxj_qdxOBfYIh2leJ2A7x85445lZhqqxDvLbmvFDPIJ__N2M8R7xfmdclR97-K1X4bJOuSTgT3pSaK5UhHEho) which we are currently using to host our docker images.

We can try to apply to their [Docker-Sponsored Open Source Program](https://www.docker.com/community/open-source/application/) which should give us a sponsored paid account. Alternatively we could switch to a different host like [ghcr](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) and use a similar approach as in #2004 to communicate the change.

WDYT?

/cc @knu

**Comment by Vad1mo** [comment]
If, for some reason, the Docker-Sponsored Open-Source Program isn't working out, we are happy to help you out. We operate a Harbor-based [container registry as a service ](https://container-registry.com/)that has many benefits over most of the other registries out there.

There are also features regarding containerized image distribution, that might be valuable too as well.

**Comment by knu** [comment]
@dsander I personally think we'd rather avoid bothering with the sponsored open source program when there are free and open alternatives like ghcr.io.

And oh, thank you for your kind offer, @Vad1mo.  We are happy to have the backers like you.  We'll see it as a good option!

**Comment by cantino** [comment]
I emailed Docker about this:

> Sounds like your organization might qualify for the Docker Sponsored Open Source (DSOS) Program.  You can find details, and the application [here](https://www.docker.com/community/open-source/application/).  Should you have questions about the program, or your eligibility, please to contact the DSOS team directly ([opensource@docker.com](mailto:opensource@docker.com)).  They are happy to help! In the event you are not eligible, I'd encourage you to contact the Docker Sales team with an inquiry around alternative pricing and discounts.  You can reach them via [sales@docker.com](mailto:sales@docker.com)

All seems pretty short sighted by Docker to me.

**Comment by dsander** [comment]
I agree with you both, even though GitHub/Microsoft could do a similar pivot in the future ghcr seems to be a good option, I have seen multiple projects switching from Docker Hub to ghcr even before their announcement. 

I'll try to look into pushing to both registries on the weekend.

It might still be worth to send a DSOS application close to their cutoff date because someone mentioned that they will not be shutting down accounts with pending applications. This might give our users a bit more time to pull the image from docker hub and get the notification about the new registry host.

**Comment by cantino** [comment]
Nice work @dsander, I'll get my instance pointing at ghcr shortly and can report back.

**Comment by cantino** [comment]
![image](https://user-images.githubusercontent.com/83835/227651414-0c3d7fe1-c64b-43bd-8060-0cd13317abb8.png)

🤦🏼

**Comment by dsander** [comment]
Yeah ... thanks for that i guess ... do we still want to stop pushing to docker hub? If not we should optimize the github actions so that the images on both repositories are identical.

**Comment by cantino** [comment]
Yea, maybe just push to both?

**Comment by virtadpt** [comment]
I think pushing to both is a good idea.  A lot of folks have already cut over entirely, and there's no telling if they'll try it again a few months later.
